python/pyspark/sql/metrics.py

```diff
@@ -98,6 +101,72 @@ def metrics(self) -> List[MetricValue]:
         return self._metrics
 
 
+class DataDebugOp:
+    """
+    The DataDebugOp class is a helper class that allows to encapsulate different reusable
+    data debug operations.
+    """
+
+    @classmethod
+    def count_values(cls) -> "DataDebugOp":
+        return DataDebugOp("count_values", F.count(F.lit(1)).alias("count_values"))
```

Comment:
So this is a wrapper of `observe` API. I think it does not simplify a lot vs the existing uscase ..

```python
observation = Observation("my metrics")
observed_df = df.observe(Observation("my metrics"), count(lit(1)).alias("count"), max(col("age")))
observation.get()
```

and this won't work for streaming.